src/main/java/LoginDelocalized/UserRepository.java

```diff
@@ -8,18 +8,33 @@ public class UserRepository {
     private List<User> users;
 
     public UserRepository() {
-
+        this.users = new ArrayList<>();
     }
 
     public void addUser(User user) {
-
+        if (!users.contains(user)) {
```

Comment:
We can still create a new user object with identical password and username to existing user.